provider/fs/src/fs_data_provider.rs

```diff
@@ -76,20 +76,27 @@ impl FsDataProvider {
         if !path.exists() {
             return Err(DataErrorKind::MarkerNotFound.with_req(marker, req));
         }
-        if !req.id.marker_attributes.is_empty() {
-            if req.metadata.attributes_prefix_match {
-                path.push(
-                    std::fs::read_dir(&path)?
-                        .filter_map(|e| e.ok()?.file_name().into_string().ok())
-                        .filter(|c| c.starts_with(req.id.marker_attributes.as_str()))
-                        .min()
-                        .ok_or(DataErrorKind::IdentifierNotFound.with_req(marker, req))?,
-                );
-            } else {
-                path.push(req.id.marker_attributes.as_str());
-            }
+        let checksum = if marker.is_singleton {
+            std::fs::read_to_string(format!("{}_checksum", path.display()))
+        } else {
+            std::fs::read_to_string(path.join(".checksum"))
         }
+        .ok()
+        .and_then(|s| s.parse().ok());
         if !marker.is_singleton {
```

Comment:
Question/Issue: Why did you move the `is_singleton` condition above the `marker_attributes` condition

Reply:
this was a bug